src/overview/epics.js

```diff
@@ -10,8 +10,14 @@ export const refreshSearchResultsUponQueryChange = action$ => action$
     .debounceTime(500) // wait until typing stops for 500ms
     .map(() => actions.refreshSearch())
 
+// When the query changed, issue the loading component
+export const playLoadingUponQueryChange = action$ => action$
+    .ofType(actions.setQuery.getType())
+    .debounceTime(500)
+    .map(() => actions.playLoading())
+    
```

Comment:
Listening to query change itself seems slightly wrong. It should listen to whether the query change started a new search. Listening to `refreshSearch` would however also trigger when the search refreshes because of database changes. Maybe `refreshSearchResultsUponQueryChange` should trigger another action (e.g. `startNewSearch`) that activates both refresh and the loading indicator, or it should just emit both those actions (with a `mergeMap`?) instead of only `refreshSearch`.